### PR TITLE
fix: parlia live sync issue

### PR DESCRIPTION
**Pull request by j75689**
### Description

1. check hash value of sealed headers to prevent incomplete headers
2. fix disconnected header checks in parlia engine
3. fix order of publishing fcu
4. print the mgas usage log for InsertDownloadedPayload action(insert block)
5. clean the cache if downloader report the header/body is invalid

### Rationale

1. need to ensure the hash of header we downloaded is expected, or may cause the following cases.

```
{"timestamp":"2024-07-24T12:11:50.792106Z","level":"DEBUG","fields":{"message":"Received wrong header","expected":"0xf758a5f65c23f82f7c3e4b74454228a1fe94232c79996f36bbcfa3484200210e","received":"0x1fdc1e2d86afaa165bfc81eecd81ae32ea561332e561e8de387743f340cee035"},"target":"downloaders"}
{"timestamp":"2024-07-24T12:11:50.792115Z","level":"DEBUG","fields":{"message":"Received wrong header","expected":"0xf758a5f65c23f82f7c3e4b74454228a1fe94232c79996f36bbcfa3484200210e","received":"0x1fdc1e2d86afaa165bfc81eecd81ae32ea561332e561e8de387743f340cee035"},"target":"downloaders"}
{"timestamp":"2024-07-24T12:11:50.792121Z","level":"DEBUG","fields":{"message":"Received wrong header","expected":"0xf758a5f65c23f82f7c3e4b74454228a1fe94232c79996f36bbcfa3484200210e","received":"0x1fdc1e2d86afaa165bfc81eecd81ae32ea561332e561e8de387743f340cee035"},"target":"downloaders"}
{"timestamp":"2024-07-24T12:11:50.792130Z","level":"DEBUG","fields":{"message":"Received wrong header","expected":"0xf758a5f65c23f82f7c3e4b74454228a1fe94232c79996f36bbcfa3484200210e","received":"0x1fdc1e2d86afaa165bfc81eecd81ae32ea561332e561e8de387743f340cee035"},"target":"downloaders"}
{"timestamp":"2024-07-24T12:11:50.792139Z","level":"DEBUG","fields":{"message":"Received wrong header","expected":"0xf758a5f65c23f82f7c3e4b74454228a1fe94232c79996f36bbcfa3484200210e","received":"0x1fdc1e2d86afaa165bfc81eecd81ae32ea561332e561e8de387743f340cee035"},"target":"downloaders"}
{"timestamp":"2024-07-24T12:11:50.792148Z","level":"DEBUG","fields":{"message":"Received wrong header","expected":"0xf758a5f65c23f82f7c3e4b74454228a1fe94232c79996f36bbcfa3484200210e","received":"0x1fdc1e2d86afaa165bfc81eecd81ae32ea561332e561e8de387743f340cee035"},"target":"downloaders"}
```

5. if the cached body/header was wrong, should clean it to prevent loop.
```
{"timestamp":"2024-07-26T12:48:53.327804Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.327890Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.327976Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.328062Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.328147Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.328231Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.328318Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
{"timestamp":"2024-07-26T12:48:53.328407Z","level":"DEBUG","fields":{"message":"Received wrong body","err":"unexpected withdrawals root","hash":"0x778c8a1d04cfa4c1e0f9da3e9435f6dec1d39c5c2d4bff521d334c7b3fffd1a5"},"target":"downloaders"}
```

### Example

n/a

### Changes

Notable changes: 
* bsc-engine

### Potential Impacts
* no

### CODE REVIEW

**Review by unclezoro — APPROVED** [review]

**Comment by forcodedancing** [comment]
Can we change https://github.com/bnb-chain/reth/blob/main/crates/bsc/engine/src/task.rs#L273 using MIN_BLOCKS_FOR_PIPELINE_RUN instead of EPOCH_SLOTS ?

**Comment by j75689** [comment]
> Can we change https://github.com/bnb-chain/reth/blob/main/crates/bsc/engine/src/task.rs#L273 using MIN_BLOCKS_FOR_PIPELINE_RUN instead of EPOCH_SLOTS ?

yes, fixed

**Review by forcodedancing — APPROVED** [review]

File: crates/storage/provider/src/providers/static_file/manager.rs [hunk 1]
```diff
@@ -637,7 +637,7 @@ impl StaticFileProvider {
                 StaticFileSegment::Sidecars => self.ensure_invariants::<_, tables::Sidecars>(
                     provider,
                     segment,
-                    highest_block,
+                    highest_tx,
```
Comment (pythonberg1997) [inline crates/storage/provider/src/providers/static_file/manager.rs]: Should not change this

Comment (j75689) [reply crates/storage/provider/src/providers/static_file/manager.rs]: reverted

**Pull request merged**